packages/machine-extractor/src/toMachineConfig.ts

```diff
@@ -10,6 +10,11 @@ import { StateNodeReturn } from './stateNode';
 import { MaybeTransitionArray } from './transitions';
 import { GetParserResult } from './utils';
 
+/**
+ * Original source code text
+ */
+export type WithFileContent<T> = T & { fileContent: string };
```

Comment:
Yeah. Because it's also used on `toConfig` and that's a public method, you'd have to provide it everywhere you use that method, including tests, etc.

Reply:
This makes sure `fileContent` is required internally while it's not present on the interface when used as a public method.

Reply:
The method is defined elsewhere - and since that method doesn't accept `fileContent` that method shouldn't include `fileContent` property. 

This type originates in this module~ so it should only be concerned about the local requirements and, in here, `fileContent` is required.

Reply:
What do you think about the improvement?